src/query/plan/operator.cpp

```diff
@@ -2564,39 +2564,76 @@ void Delete::DeleteCursor::UpdateDeleteBuffer(Frame &frame, ExecutionContext &co
     expression_results.emplace_back(expression->Accept(evaluator));
   }
 
+#ifdef MG_ENTERPRISE
+  auto vertex_auth_checker = [&context](const VertexAccessor &va) {
+    if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
+        !context.auth_checker->Has(va, storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE)) {
+      throw QueryRuntimeException("Vertex not deleted due to not having enough permission!");
+    }
+  };
+
+  auto edge_auth_checker = [&context](const EdgeAccessor &ea) {
+    if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
+        !(context.auth_checker->Has(ea, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE) &&
+          context.auth_checker->Has(ea.To(), storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::UPDATE) &&
+          context.auth_checker->Has(ea.From(), storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::UPDATE))) {
+      throw QueryRuntimeException("Edge not deleted due to not having enough permission!");
+    }
+  };
+
+#endif
+
   for (TypedValue &expression_result : expression_results) {
     AbortCheck(context);
     switch (expression_result.type()) {
       case TypedValue::Type::Vertex: {
         auto va = expression_result.ValueVertex();
 #ifdef MG_ENTERPRISE
-        if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
-            !context.auth_checker->Has(va, storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE)) {
-          throw QueryRuntimeException("Vertex not deleted due to not having enough permission!");
-        }
+        vertex_auth_checker(va);
 #endif
         buffer_.nodes.push_back(va);
         break;
       }
       case TypedValue::Type::Edge: {
         auto ea = expression_result.ValueEdge();
 #ifdef MG_ENTERPRISE
-        if (license::global_license_checker.IsEnterpriseValidFast() && context.auth_checker &&
-            !(context.auth_checker->Has(ea, query::AuthQuery::FineGrainedPrivilege::CREATE_DELETE) &&
-              context.auth_checker->Has(ea.To(), storage::View::NEW, query::AuthQuery::FineGrainedPrivilege::UPDATE) &&
-              context.auth_checker->Has(ea.From(), storage::View::NEW,
-                                        query::AuthQuery::FineGrainedPrivilege::UPDATE))) {
-          throw QueryRuntimeException("Edge not deleted due to not having enough permission!");
-        }
+        edge_auth_checker(ea);
 #endif
         buffer_.edges.push_back(ea);
         break;
       }
+      case TypedValue::Type::Path: {
+        auto path = expression_result.ValuePath();
+        std::transform(path.edges().cbegin(), path.edges().cend(), std::back_inserter(buffer_.edges),
```

Comment:
Did some quick benchmarking and it seems like doing a loop to check auth and then just copying the whole vector is faster than transform.

Reply:
We could just directly copy in case MG_ENTERPRISE OFF, since we support path deletion, but simply do not check authentication in community.

Reply:
can you show me that benchmark, not sure what you mean?